### PR TITLE
Add support for CONDUIT_PROMETHEUS_LABELS to the proxy

### PR DESCRIPTION
**Pull request by hawkw**
This PR adds support for the `CONDUIT_PROMETHEUS_LABELS` environment variable in the proxy. I've added a unit test asserting that the labels in the env var are correctly added to metrics.

Closes ##618.

### CODE REVIEW

**Comment by hawkw** [comment]
I'm not sure if this feature is still required after adding #605, but @olix0r suggested that it might be worth having regardless.

File: proxy/src/config.rs [hunk 1]
```diff
@@ -240,6 +247,8 @@ impl<'a> TryFrom<&'a Strings> for Config {
             pod_name: pod_name?,
             pod_namespace: pod_namespace?,
             node_name: node_name?,
+            prometheus_labels: prometheus_labels?
+                .map(Arc::from),
```
[the remark this answers is not in the document]
Comment (hawkw) [reply proxy/src/config.rs]: there's an `Arc<str>::from<String>` impl that also turns the string into a ref-counted string slice.

File: proxy/src/telemetry/control.rs [hunk 2]
```diff
@@ -24,6 +24,8 @@ pub struct MakeControl {
     flush_interval: Duration,
 
     process_ctx: Arc<ctx::Process>,
+
+    prometheus_labels: Option<Arc<str>>,
```
Comment (olix0r) [inline proxy/src/telemetry/control.rs]: do you think it makes sense to just store this in the process ctx? nearly everywhere we reference it, we also have a `process_ctx` already.

**Comment by olix0r** [comment]
We've decided to hold off on this until there's a concrete use case.

**Pull request closed without merging**